gi-gtk-declarative-app-simple/src/GI/Gtk/Declarative/App/Simple.hs

```diff
@@ -135,12 +139,12 @@ runLoop App {..} = do
         -- If the action returned by the update function produced an event, then
         -- we write that to the channel.
         --
-        -- TODO: Use prioritized queue for events returned by 'update', to take
-        -- precendence over those from 'inputs'.
-        void . forkIO $ action >>= maybe (return ()) (writeChan events)
-
-        -- Finally, we loop.
-        loop newState newMarkup events sub newModel
+        snd <$> Async.concurrently
```

Comment:
I'm no expert on Haskell concurrency, but my naive guess from looking at this is that every `loop` iteration would result in another level being added to an ever-growing stack of in-complete executions of `Async.concurrently` (since none of the executions can finish until `loop` finishes). So this would cause a memory leak.

Is this a valid concern?

Reply:
Yes, good point. I totally missed that one. Each iteration will increase the stack.

I need to find another solution. I do really thank you for catching that.

Reply:
I think I can use https://hackage.haskell.org/package/async-2.2.2/docs/Control-Concurrent-Async.html#v:link to link the exception with the current thread.

Reply:
:+1: 

Possibly `mapConcurrently` is another option. I.e. `loop` could produce a list that is passed to `mapConcurrently`. Off the top of my head I'm not 100% sure this would be safe from leaks either, though.